Externals.cfg

```diff
@@ -17,8 +17,8 @@ required = True
 protocol = git
 repo_url = https://github.com/ufs-community/ufs-weather-model
 # Specify either a branch name or a hash but not both.
-branch = release/public-v2
-#hash = 8165575
```

Comment:
@climbfuji Is this how the MRW App handles their Externals.cfg file for the release?  Since we're still under rapid-development (even for bug fixes for the release), I'm concerned that this change will require us to constantly update Externals.cfg to stay up to date.  

Note that we've had discussions about creating a stable branch of regional_workflow that would contain specific hashes in the Externals.cfg file, while the develop branch would point to the HEAD of the branch.  New hashes would then have to be extensively tested before they're updated in the Externals.cfg file for the stable branch, while the develop branch would always point to the HEAD and users would be made aware that it may not work.

Reply:
In my opinion a certain hash of the srweather-app code is guaranteed to work with exactly one hash of each of its submodules / externals, everything else is gambling. Yes, everytime you update one of the submodules you need to test it with the app and as part of the testing update the hash. The PR that is created to update it documents the successful testing with those particular hashes.

Reply:
But note that you don't have to update to a new hash every time, you can do that whenever necessary and as such accumulate a few changes to the submodules (although this increases the risk to break stuff).

Reply:
OK, that makes sense.  I see that the ufs-mrweather-app points to tags for each repo in the Externals.cfg file.  I'm assuming those correspond to specific hashes, so that would be equivalent.

Reply:
Yes, tags and hashes are equivalent in a sense that they are static; tags however can be updated by force to a new hash (re-tagging) - not recommended unless absolutely necessary.

Reply:
Should I also change the branches to hashes for EMC_post and UFS_UTILS?

Reply:
Yes, if you don't mind, that would be great.  I'll approve right afterward!

Reply:
Done. I'll try to check out the PR again to make sure I got all the hashes correct.

Reply:
OK, I'll preliminarily approve this PR.

Reply:
The hashes do match the head of the branches at this point in time, so all good.

Reply:
OK, merging!